<a href="https://colab.research.google.com/github/Ekene-Ebenezer/Gomycode-Data-Science/blob/main/ydata_profiling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Ydata Profiling 2

Checkpoint Objective

In this checkpoint, we are going to work on 'The Tunisian flight company: Tunisair' dataset.

Dataset description : This dataset include historical data of flight delays. the objective is to investigate the flight delays factors.

[Datalink](https://drive.google.com/file/d/1W7WFOiCZu9qFeKFhM0_1bV7BQWsgSuMv/view?usp=sharing)



**Instructions**

**Part 1: Data Exploration with Pandas**

1. Load the provided dataset into a pandas dataframe
Use pandas to explore the dataset. For example, you might start by using the head() and info() methods to get an overview of the data.

2. Look for missing values in the dataset. You can use the isnull() method to identify missing values.

3. Use pandas to calculate some summary statistics for the dataset. For example, you might use the describe() method to get summary statistics for the numerical columns in the dataset.

**Part 2: Data Exploration with ydata-Profiling**

1. Use ydata-profiling to generate a report of the provided dataset.
2. Look for missing values in the dataset. You can use the report generated by ydata-profiling to identify missing values.
3. Look for correlations between different columns in the dataset. You can use the report generated by ydata-profiling to identify correlations between different columns.
4. Identify any outliers or unusual values in the dataset. You can use the report generated by ydata-profiling to identify any outliers or unusual values.

**Summary**

At the end of this exercise, write a summary of your findings. Did you find any interesting patterns or correlations in the data? Were there any issues or challenges you encountered while exploring the dataset? Use your summary to reflect on your experience using pandas and ydata-profiling to explore and understand a new dataset.




# - Data Explorations with Pandas

---



In [ ]:
#install Ydata Profiling
!pip install pandas-profiling

In [ ]:
import pandas as pd


In [ ]:
#importing dataset
Tunisia_flights_dataset = pd.read_csv("/content/Tunisair_flights_dataset.csv")

In [ ]:
#basic information about the dataset
Tunisia_flights_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 107833 entries, 0 to 107832
Data columns (total 9 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   Filght_date               107833 non-null  object 
 1   Flight_ID                 107833 non-null  object 
 2   Departure point           107833 non-null  object 
 3   Arrival point             107833 non-null  object 
 4   Scheduled_departure_time  107833 non-null  object 
 5   Scheduled_arrival_time    107833 non-null  object 
 6   STATUS                    107833 non-null  object 
 7   Aircraft_code             107833 non-null  object 
 8   Arrival delay             107833 non-null  float64
dtypes: float64(1), object(8)
memory usage: 7.4+ MB


In [ ]:
Tunisia_flights_dataset.head()

Filght_date Flight_ID Departure point Arrival point  \
0  2016-01-03  TU 0712              CMN           TUN   
1  2016-01-13  TU 0757              MXP           TUN   
2  2016-01-16  TU 0214              TUN           IST   
3  2016-01-17  TU 0480              DJE           NTE   
4  2016-01-17  TU 0338              TUN           ALG   

  Scheduled_departure_time Scheduled_arrival_time STATUS Aircraft_code  \
0      2016-01-03 10:30:00    2016-01-03 12.55.00    ATA     TU 32AIMN   
1      2016-01-13 15:05:00    2016-01-13 16.55.00    ATA     TU 31BIMO   
2      2016-01-16 04:10:00    2016-01-16 06.45.00    ATA     TU 32AIMN   
3      2016-01-17 14:10:00    2016-01-17 17.00.00    ATA     TU 736IOK   
4      2016-01-17 14:30:00    2016-01-17 15.50.00    ATA     TU 320IMU   

   Arrival delay  
0          260.0  
1           20.0  
2            0.0  
3            0.0  
4           22.0

In [ ]:
Tunisia_flights_dataset.isnull().sum()

Filght_date                 0
Flight_ID                   0
Departure point             0
Arrival point               0
Scheduled_departure_time    0
Scheduled_arrival_time      0
STATUS                      0
Aircraft_code               0
Arrival delay               0
dtype: int64

In [ ]:
#Descriptive statistics of the dataset

Tunisia_flights_dataset.describe()

Arrival delay
count  107833.000000
mean       48.733013
std       117.135562
min         0.000000
25%         0.000000
50%        14.000000
75%        43.000000
max      3451.000000

# - Data Exploration with Ydata

In [ ]:
# Import the Ydata_profiling library that have been installed
import ydata_profiling as ydp

# Imports profile for creating html or notebook from ydata_profiling results
import profile

#
from ydata_profiling import ProfileReport

In [ ]:
#Converts the profile created into a notebook
Tunisia_flights_DataProfile.to_notebook_iframe()

This cell output is too large and can only be displayed while logged in.


In [ ]:
#Converts the profile created into a html file
Tunisia_flights_DataProfile.to_file("Edu_DataProfile")

/usr/local/lib/python3.10/dist-packages/ydata_profiling/profile_report.py:354: UserWarning: Try running command: 'pip install --upgrade Pillow' to avoid ValueError
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/ydata_profiling/profile_report.py:375: UserWarning: Extension  not supported. For now we assume .html was intended. To remove this warning, please use .html or .json.
  warnings.warn(


Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

# Summary of the Dataset Profiling

---

Based on the result from the Ydata Profiling;
>There are no missing or null values among the 107,833 observations in the dataset.

>All columns, except for the arrival time column, are of the object data type.

>Ydata has flagged a concern regarding the number of zeros in the dataset, specifically 38,168 (35.4%) zeros. These zeros are attributed to the arrival date column, suggesting that Tunisia flights have achieved a remarkable 35.4% zero-delay rate, aligning with the standard practice of minimizing delays for airline customers.

>The arrival delay column exhibits a maximum delay of 3,451 minutes, indicating the presence of outliers. This outlier is also evident in the mean delay arrival time of 48 minutes, with a median delay of 14 minutes. This suggests that while the majority of flights experience relatively low delays, there are instances of significantly longer delays, contributing to the high mean delay.



